processing/src/main/java/org/apache/druid/utils/ConnectionUriUtils.java

```diff
@@ -177,21 +182,75 @@ public static Set<String> tryParsePostgresConnectionUri(String connectionUri)
   }
 
   public static Set<String> tryParseMySqlConnectionUri(String connectionUri)
-      throws ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException,
+      throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException,
              InvocationTargetException
   {
-    Class<?> driverClass = Class.forName(MYSQL_NON_REGISTERING_DRIVER);
-    Method parseUrl = driverClass.getMethod("parseURL", String.class, Properties.class);
-    // almost the same as postgres, but is an instance level method
-    Properties properties = (Properties) parseUrl.invoke(
-        driverClass.getConstructor().newInstance(),
-        connectionUri,
-        null
-    );
+    Class<?> connectionUrlClass = Class.forName(MYSQL_CONNECTION_URL);
```

Comment:
can you describe your logic in a bit more detail?

Reply:
Now the method 
- Checks if the string is in acceptable format 
- Get the `list<hostInfo>` obtained from the string
- From the `hostInfo` obtain the properties and return the key of each properties